src/Support/AliasedRelationships/ModelRelationshipAdder.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rebing\GraphQL\Support\AliasedRelationships;
+
+use Illuminate\Database\Eloquent\Model;
+
+class ModelRelationshipAdder
+{
+    /**
+     * @param class-string<Model> $model
+     * @param array<string,string> $relationships
+     * @return void
+     */
+    public static function add(string $model, array $relationships): void
+    {
+        $model::addGlobalScope(new AddModelRelationshipScope($relationships));
```

Comment:
👀 

Isn't that somehow dangerous (side-effects), permanently mutating the the models global scopes?

Reply:
> I don't feel confident doing a good review on this one.
> 
> This shows the absurd complexity we've to put upon us because of the concept of the `SelectFields` and once more shows me this approach goes into the wrong direction somehow…
> 
> Leads me to believe `SelectFields` should be a standalone component and not in this library TBH 😄
> 
> Mutating models scopes, adding macros to builders .. this somehow looks dangerous to me 🤷‍♀️
> 
> When I think about I would implement a dedicated PHP server with this library (i.e. never restart, a daemon) I can foresee thus causing problems. Is there no other way to solve this aliasing / double-query problem?
> 
> Not really sure how to move forward with this one TBH.
> 
> > Leads me to believe `SelectFields` should be a standalone component and not in this library TBH 😄
> 
> To clarify this, I believe the proper approach is to use a dataloader concept to solve these things in general (that's what I'm doing in my projects, I'm not using SelectFields)

I love the SelectFields and use it for 99% of my queries, makes it so easy to set up

Reply:
> 👀
> 
> Isn't that somehow dangerous (side-effects), permanently mutating the the models global scopes?

yes it is definitely a side effect, but since SelectFields are made to be used with eloquent's ->with(). The only way to fix this is to generate the relationships.

The side effect is minimized by generating relationships with **_generated_** in front of all generated relationships, so I would say this makes it as safe as possible.

Reply:
yeah we might need to tore the relationship down if some one uses the same relationship again